### PR TITLE
Extract spanner tests into grpcgcp_tests package

### PR DESCRIPTION
**Pull request by nimf**
Extracted spanner tests into separate package not to depend on `spanner` and `genproto` packages.

- Moved unique tests from `spanner_pb_test.go` to `gcp_balancer_test.go` and `gcp_picker_test.go`
- Removed the rest of `spanner_pb_test.go` as these test were duplicating other tests.

### CODE REVIEW

File: grpcgcp/gcp_balancer_test.go [hunk 1]
```diff
@@ -382,11 +518,15 @@ func TestRoundRobinForBind(t *testing.T) {
...
+	// Bring other subconns to ready.
+	b.UpdateSubConnState(scs[0], balancer.SubConnState{ConnectivityState: connectivity.Ready})
+	b.UpdateSubConnState(scs[2], balancer.SubConnState{ConnectivityState: connectivity.Ready})
+
 	// Create more regular calls to reach the limit to spawn new subconn.
-	for i := 0; i < 7; i++ {
-		pr, err := b.picker.Pick(balancer.PickInfo{FullMethodName: "dummyService/noAffinity", Ctx: context.TODO()})
-		if want := scs[1]; pr.SubConn != want || err != nil {
-			t.Fatalf("gcpPicker.Pick returns %v, %v, want: %v, nil", pr.SubConn, err, want)
+	for i := 0; i < 24; i++ {
```
Comment (fengli79) [inline grpcgcp/gcp_balancer_test.go:526]: better to define it as a const.

File: grpcgcp/gcp_balancer_test.go [hunk 2]
```diff
@@ -165,6 +226,81 @@ func TestCreatesMinSubConns(t *testing.T) {
...
+				ChannelPool: &pb.ChannelPoolConfig{
+					MinSize:                          3,
+					MaxSize:                          5,
+					MaxConcurrentStreamsLowWatermark: 1,
+				},
+			},
+		},
+	})
+
+	if want := 3; len(b.scRefs) != want {
```
Comment (fengli79) [inline grpcgcp/gcp_balancer_test.go:259]: try avoid using such magic numbers and define them as const, with necessary additive values.

**Review by fengli79 — APPROVED** [review]

File: grpcgcp/gcp_balancer_test.go (hunk 2 above)
Comment (nimf) [reply grpcgcp/gcp_balancer_test.go]: Yeah, I was thinking about that. Done.

File: grpcgcp/gcp_balancer_test.go (hunk 1 above)
Comment (nimf) [reply grpcgcp/gcp_balancer_test.go]: Done

**Pull request merged**